docs/tutorials/go-built-in.md

```diff
@@ -56,338 +56,495 @@ We'll start by creating a new Go project.
+Finally, make sure to add the bytes package to the `import` stanza at the top of `app.go`:
 
 ```go
-import "github.com/dgraph-io/badger"
+import(
+	"bytes"
 
-type KVStoreApplication struct {
-	db           *badger.DB
-	currentBatch *badger.Txn
-}
-
-func NewKVStoreApplication(db *badger.DB) *KVStoreApplication {
-	return &KVStoreApplication{
-		db: db,
-	}
-}
+	"github.com/dgraph-io/badger/v3"
+	abcitypes "github.com/cometbft/cometbft/abci/types"
+)
 ```
 
-### 1.3.2 BeginBlock -> DeliverTx -> EndBlock -> Commit
 
-When Tendermint Core has decided on the block, it's transfered to the
-application in 3 parts: `BeginBlock`, one `DeliverTx` per transaction and
-`EndBlock` in the end. DeliverTx are being transfered asynchronously, but the
-responses are expected to come in order.
+### 1.3.3 BeginBlock -> DeliverTx -> EndBlock -> Commit
+
+When the CometBFT consensus engine has decided on the block, the block is transferred to the
+application over three ABCI method calls: `BeginBlock`, `DeliverTx`, and `EndBlock`.
+
+- `BeginBlock` is called once to indicate to the application that it is about to
+receive a block.
+- `DeliverTx` is called repeatedly, once for each application transaction that was included in the block.
+- `EndBlock` is called once to indicate to the application that no more transactions
+will be delivered to the application in within this block.
+
+Note that, to implement these calls in our application we're going to make use of Badger's
+transaction mechanism. We will always refer to these as Badger transactions, not to
+confuse them with the transactions included in the blocks delivered by CometBFT,
+the _application transactions_.
+
+First, let's create a new Badger transaction during `BeginBlock`. All application transactions in the
+current block will be executed within this Badger transaction.
+Then, return informing CometBFT that the application is ready to receive application transactions:
 
 ```go
 func (app *KVStoreApplication) BeginBlock(req abcitypes.RequestBeginBlock) abcitypes.ResponseBeginBlock {
-	app.currentBatch = app.db.NewTransaction(true)
+	app.onGoingBlock = app.db.NewTransaction(true)
 	return abcitypes.ResponseBeginBlock{}
 }
-
 ```
 
-Here we create a batch, which will store block's transactions.
+Next, let's modify `DeliverTx` to add the `key` and `value` to the database transaction every time our application
+receives a new application transaction through `RequestDeliverTx`.
 
 ```go
 func (app *KVStoreApplication) DeliverTx(req abcitypes.RequestDeliverTx) abcitypes.ResponseDeliverTx {
-	code := app.isValid(req.Tx)
-	if code != 0 {
+	if code := app.isValid(req.Tx); code != 0 {
 		return abcitypes.ResponseDeliverTx{Code: code}
 	}
 
-	parts := bytes.Split(req.Tx, []byte("="))
+	parts := bytes.SplitN(req.Tx, []byte("="), 2)
 	key, value := parts[0], parts[1]
 
-	err := app.currentBatch.Set(key, value)
-	if err != nil {
-		panic(err)
+	if err := app.onGoingBlock.Set(key, value); err != nil {
```

Comment:
```suggestion
	if err := app.currentBatch.Set(key, value); err != nil {
```

Reply:
I think it is a local variable, right? Defined in line 232.